File: 68주차/피로도.py
answer = 0
def solution(k,dungeons):
    n = len(dungeons)
    visited = [0]*n
    def dfs(L,remain):
        global answer
        if L > n:
            return
        answer = max(answer,L)
        for i in range(n):
            if not visited[i] and remain >= dungeons[i][0]:# 최소 필요 피로도 체크.
                visited[i] = 1
                dfs(L+1, remain-dungeons[i][1])# 소모 피로도 반영
                visited[i] = 0
    dfs(0,k)
    return answer

File: 94주차/피로도.js
function solution(k, dungeons) {
  var answer = 0;
  const n = dungeons.length;
  const visited = Array(n).fill(0);

  const dfs = (now, L) => {
    answer = Math.max(answer, L);
    for (let i = 0; i < n; i++) {
      if (!visited[i] && now >= dungeons[i][0]) {
        visited[i] = 1;
        dfs(now - dungeons[i][1], L + 1);
        visited[i] = 0;
      }
    }
  };
  dfs(k, 0);
  return answer;
}
// 프로그래머스 문제 피로도.js
